### PR TITLE
Adding custom value option to checkboxes interface

### PR DESCRIPTION
**Pull request by sethjgore**
The diff is bloated due to unintended syntax formatting changes to the files. My editor has automatic syntax formatting, but I'm happy to follow whatever is the syntax spec for Directus.

### CODE REVIEW

**Comment by sethjgore** [comment]
**interface.js**

refactored `updateValue` which extracted value, delimited the value, and setting to the model into separate functions, 

Now, resepctively, they are`value`, `delimitize`, and `update` which combines `value` and `customValue` into one array for updating

**main.css**
enable proper `input.small` width (previously min-width of input would prevent it from going smaller than 400px)

**component.js**
added `allow_custom_value' option that will enable or disable custom single-value text input

**Comment by rijkvanzanten** [comment]
Hey @sethjgore, looks nice! We might want to keep an eye on https://github.com/directus/directus/pull/1762 to prevent conflicts as it aims to fix the min width issue.

**Comment by sethjgore** [comment]
@rijkvanzanten thanks for reviewing. Definitely will keep an eye on that.

**Comment by rijkvanzanten** [comment]
@sethjgore I've merged the text-input size fix so we can focus on getting this branch perfect now 🙂 
You already mentioned your editor auto-formatting the syntax. Directus uses [EditorConfig](http://editorconfig.org) (rules: [`.editorconfig`](https://github.com/directus/directus/blob/master/.editorconfig)) for formatting and [`xo`](https://github.com/sindresorhus/xo) (rules: [`.package.json`](https://github.com/directus/directus/blob/master/package.json#L18)) for linting the JavaScript. 
Could you maybe checkout if your editor has plugins for these?

**Comment by sethjgore** [comment]
Great, thanks for the information. Applying..

**Comment by WellingGuzman** [comment]
Hey @sethjgore is this PR done and ready?

**Comment by sethjgore** [comment]
It's ready, sans the ridiculous lines of "mis-syntaxed" code. 

I still have to figure out how to get Sublime Editor 3 to respect `.editorconfig' along with other syntax packages i use for most of my projects.

**Comment by sethjgore** [comment]
Removing the main.css modifications, should pass the tests.

**Comment by rijkvanzanten** [comment]
@sethjgore Maybe these work?

https://github.com/sindresorhus/editorconfig-sublime
https://github.com/sindresorhus/SublimeLinter-contrib-xo

File: app/core/interfaces/checkboxes/interface.js [hunk 1]
```diff
@@ -15,50 +16,91 @@ define(['core/UIView', 'underscore'], function (UIView, _) {
...
+    },
+    customValue: function () {
+
+      var customValues = this.$('#customText')[0].value;
+
+      return customValues;
+    },
+
+    update: function () {
+      var out = []
```
Comment (WellingGuzman) [inline app/core/interfaces/checkboxes/interface.js]: I know it's not clear, but we use commas, and there's some missing commas.

File: app/core/interfaces/checkboxes/interface.js [hunk 2]
```diff
@@ -15,50 +16,91 @@ define(['core/UIView', 'underscore'], function (UIView, _) {
...
+
+      out.push(this.value())
+
+      out.push(this.customValue())
```
Comment (WellingGuzman) [inline app/core/interfaces/checkboxes/interface.js]: We should check if the value exists before pushing it into the list, to prevent empty elements.

**Review by wellingguzman — APPROVED** [review]
I'm going to fix the things I mentioned ~above~ below.

File: app/core/interfaces/checkboxes/interface.js (hunk 1 above)
Comment (sethjgore) [reply app/core/interfaces/checkboxes/interface.js]: Ok, thanks

**Pull request merged**